### ISSUE TITLE
Add `vinxi test`

### ISSUE THREAD

**Issue by atk** [nksaraf/vinxi#174]
Currently, testing needs to be set up manually, since we use the vite config for our own configuration, which does not really work with vitest. It would also be useful to be able to configure which parts of the system should be available during testing (static assets, ssr, client, etc.). In order to do this, we should also wrap vitest (as the preferred way to test vinxi apps and optional peer dependency) in a `vinxi test` cli functionality.

**Comment by nksaraf** [comment]
Yeah okay let me put my ideas down and I think I probably need your help sifting through them.

Approach 1: 

I expose a `vinxi test` command that knows how to run vitest tests and maybe playwright tests for integration testing. Will probably need to figure out how the file can tell us which env it needs to run in. Because the difference is not just vitest/playwright, its also server/client, etc. So a specific test file should annotate itself as to how it needs to be run,

eg. 
```ts
// @test-runner vitest
// @test-env ssr

import {} from "vinxi/test"
```

```ts
// @test-runner playwright

import {} from "vinxi/test"
```

Approach 2:

I expose a `defineVitestConfig` exported from `vinxi/test` that can be used in a `vitest.config.ts` to derive a vitest config out of vinxi app config. It will take options to drive the config, and then will also probably need to respond to environmet variables. Though then you can probably use `vitest` to run your tests. 

```ts
import { defineVitestConfig } from "vinxi/test"
import app from './app.config.js'

export default defineVitestConfig(app)
```

**Comment by atk** [comment]
Let me add a few thoughts:

* You may want to expose other routers to the tests based on global and per-file configuration, e.g. `// @vinxi-routers ssr, client, public`
* Even if you have a vinxi test command, you can run normal vite tests by temporarily creating a config for vitest, so you don't need to export anything except the CLI command
* I would very much prefer as few extra configuration for testing as necessary; ideally, tests should run out of the box without any extra configuration

**Comment by ben-pr-p** [comment]
Approach 2 looks good to me.

I believe this is blocking testing server functions in tanstack/start, if I'm not mistaken.

Do you know of any work-around in the meantime?

**Comment by atk** [comment]
If you have pure front end components, you can still unit test them without solid start and use playwright for the integration tests.

**Comment by ben-pr-p** [comment]
Yes of course - I’m hoping to write unit tests for my server functions as well.

**Comment by figitaki** [comment]
Is there a good example of how to setup testing manually using `vitest.config.ts` in the meantime?
